src/ui/src/containers/live-widgets/table/query-result-viewer.tsx

```diff
@@ -122,7 +122,7 @@ export interface QueryResultTableProps {
 }
 
 export const QueryResultTable = React.memo<QueryResultTableProps>(({
-  display, table, propagatedArgs, customGutters = [], setExternalControls,
```

Comment:
Part of the performance issue was that this array had a different identity on every render, falsely causing several components to re-render. By letting it stay undefined, this stops.